src/kv/store.h

```diff
@@ -409,6 +472,23 @@ namespace kv
       for (auto& map : maps)
         map.second->rollback(v);
 
+      // TODO: Lock dynamic maps? Or perhaps flatten both into a single
+      // container?
```

Comment:
`map->rollback(v)` expects the map to be locked beforehand so I would say yes, definitely.